gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/MostlyMySqlDagManagementStateStore.java

```diff
@@ -0,0 +1,200 @@
+  }
+
+  @Override
+  public void deleteFailedDag(DagManager.DagId dagId) throws IOException {
+    this.failedDagStateStore.cleanUp(dagId.toString());
+  }
+
+  @Override
+  public List<Dag<JobExecutionPlan>> getDags() throws IOException {
+    return this.dagStateStore.getDags();
+  }
+
+  @Override
+  public Optional<Dag<JobExecutionPlan>> getFailedDag(DagManager.DagId dagId) throws IOException {
+    return Optional.of(this.failedDagStateStore.getDag(dagId.toString()));
+  }
+
+  @Override
+  public Set<String> getFailedDagIds() throws IOException {
+    return this.failedDagStateStore.getDagIds();
+  }
+
+  @Override
+  // todo - updating different mapps here and in addDagNodeState can result in inconsistency between the maps
+  public synchronized void deleteDagNodeState(DagManager.DagId dagId, Dag.DagNode<JobExecutionPlan> dagNode) {
+    this.jobToDag.remove(dagNode);
+    this.dagNodes.remove(dagNode.getValue().getId());
+    this.dagToDeadline.remove(dagId);
+    this.dagToJobs.get(dagId).remove(dagNode);
+    if (this.dagToJobs.get(dagId).isEmpty()) {
+      this.dagToJobs.remove(dagId);
+    }
+  }
+
+  // todo - updating different mapps here and in deleteDagNodeState can result in inconsistency between the maps
+  @Override
+  public synchronized void addDagNodeState(Dag.DagNode<JobExecutionPlan> dagNode, DagManager.DagId dagId)
+      throws IOException {
+    Optional<Dag<JobExecutionPlan>> dag = getDag(dagId);
+    if (!dag.isPresent()) {
+      throw new RuntimeException("Dag " + dagId + " not found");
+    }
+    this.jobToDag.put(dagNode, dag.get());
+    this.dagNodes.put(dagNode.getValue().getId(), dagNode);
+    if (!this.dagToJobs.containsKey(dagId)) {
+      this.dagToJobs.put(dagId, Lists.newLinkedList());
+    }
+    this.dagToJobs.get(dagId).add(dagNode);
+  }
+
+  @Override
+  public Optional<Dag<JobExecutionPlan>> getDag(DagManager.DagId dagId) throws IOException {
+    return Optional.of(this.dagStateStore.getDag(dagId.toString()));
+  }
+
+  @Override
+  public boolean containsDag(DagManager.DagId dagId) throws IOException {
+    return this.dagStateStore.existsDag(dagId);
+  }
+
+  @Override
+  public Optional<Dag.DagNode<JobExecutionPlan>> getDagNode(DagNodeId dagNodeId) {
+    return Optional.of(this.dagNodes.get(dagNodeId));
+  }
+
+
+  @Override
+  public Optional<Dag<JobExecutionPlan>> getParentDag(Dag.DagNode<JobExecutionPlan> dagNode) {
+    return Optional.of(this.jobToDag.get(dagNode));
+  }
+
+  @Override
+  public List<Dag.DagNode<JobExecutionPlan>> getDagNodes(DagManager.DagId dagId) {
+    List<Dag.DagNode<JobExecutionPlan>> dagNodes = this.dagToJobs.get(dagId);
+    if (dagNodes != null) {
+      return dagNodes;
+    } else {
+      return Lists.newLinkedList();
+    }
+  }
+
+  public void initQuota(Collection<Dag<JobExecutionPlan>> dags) {
+    // This implementation does not need to update quota usage when the service restarts or when its leadership status changes
```

Comment:
mark with todo or unimplemented exception

Reply:
no it is not "unimplemented" , it is not "needed" for this implementation